### DISCUSSION TITLE
Welcome to Anura3D_OpenSource Discussions!

### DISCUSSION THREAD

**Discussion by murphy-john** [Anura3D/Anura3D_OpenSource#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ZeidBit** [comment]
Hi everyone, 

My name is Zeid Bitar and I'm a geotechincal engineer at AECOM. I am interested in further utilizing MPM methods for slope run-out problems. My question is about how to implement a phreatic surface when creating slope geometries to obtain the correct effective stress formulation. In the current version of Anura3D I don't see the option to do so. What could be recommended workaround?

**Reply by murphy-john** [reply]
Hi Zeid,
Great to see your interest in Anura3D and willingness to get our discussions started! 

There's no way to directly specify the phreatic surface in Anura3D in the same way as other geotechnical software.  

Assuming a 2D model, I think the fastest way is to create a geometry and specify lines along the assumed/known phreatic surface. Then you'd specify the material in the GiD surface above as a "dry material" and the material below the line as one of the saturated soil models. You'd then have to apply the appropriate boundary conditions and allow the flow and stresses in the model to equilibrate. This could be accomplished with the quasi-static convergence calculation. The quasi-static convergence calculation is used in a couple of the tutorials. You could follow those examples to see how it works, then adapt it to your solution.

I'll add that the code is still a "research code" and is being actively developed by our community. Also, we are just now transitioning to Github with a soft launch. We aren't officially launching the page until later this month. That being said, I'm hopeful someone else can provide a better approach for you.

**Reply by ZeidBit** [reply]
Hey John, 

Thanks for the reply! It seems your proposed method is the best way forward. I found some research papers on landslides but they have been submerged problems (coupled formulation). It is interesting to see how the phreatic surface can be incorporated in the future. Thanks again!

**Reply by ayerrocolom** [reply]
In the new version of the Anura3D (v 2022), you can prescribe an initial phreatic surface. Please, check chapter 11 of the tutorial manual v 2022 [here](https://github.com/Anura3D/Anura3D_OpenSource/blob/main/manuals/TutorialManual_2022.pdf).

**Comment by HusterAGG** [comment]
Hi everyone,

My name is Shen and I'm a geotechincal PhD student. I am also interested in utilizing MPM methods for slope run-out problems. In my calculation, there is an error showing as 【ERROR】：<time step: xx>Failed finding new element for particle xx in element xx at position xx xx xx in time step xx. I wonder the possible reasons for this issue. Thanks a lot!

**Reply by FrancescaCeccato** [reply]
Dear Shen, 
there are many reasons why it cannot find a new element for a material point. The first things to check are the correct input of boundary conditions and stress state. Did you check the output? what is happeing?

**Reply by ZeidBit** [reply]
Hey Shen, 

Sorry couldn't reply to your message since I deleted the comment. Thanks for your suggestion, but I realized that one of my solid density parameters was missing a digit leading to unrealistic flow of material. In regards to your question, I would suggest rechecking the fixities especially along the base to not allow it to move out of the mesh. I am using GiD version 15.0.4 Anura3D 2019.1. I am sorry I am not familiar with using cylindrical coordinates as of yet. Best of luck!

**Comment by zhangxiaopengY** [comment]
Hi everyone,
My name is Zhang, a geotechincal PhD student. I am interested in coseismic landslide simulation with MPM. I set a model of a landslide body on a rigid slab, when I initial the stress, the interpenetration happen at the bottom (the contact algorithm has been enabled), and there is an error: [ERROR] : (time step:   7295)   Particle    932 from element     94 lies outside the mesh. Can you give me some suggestions, and what could be the cause? Thanks a lot!

**Reply by FrancescaCeccato** [reply]
I suggest to initialize the stresses with K0 procedure or run a quasi-static gravity loading phase without contact algorithm. You can activate the contact at the second load step together with seismic loading.

**Reply by zhangxiaopengY** [reply]
Ok, thanks a lot. If I set a rigid plate under the landslide body, it must switch on the contact, how I can deal it.

**Reply by FrancescaCeccato** [reply]
Rigid body only works with contact algorithm. A workaournd could be using a linear elastic material at the base. Are you appling a velocity to this base to simulate the seismic loading? If so, then you can set the velocity to 0 at the beginning and it will behave as a rigid body.
Maybe also K0 can help at the beginning

**Reply by zhangxiaopengY** [reply]
Ok, I will try it. I use the prescribed velocity to simulate the seismic loading after the initialize the stresses. I found that I change the base to linear elastic it work, but with rigid body, it not. I also test the block sample in the tutorial manual with a rigid base, the same problems also emerge (MP interpenetration, and MP moving out of the calculation mesh). I don'n know why?

**Reply by murphy-john** [reply]
Hi @zhangxiaopengY, were you able to find a solution? You might have to use the debug feature in Visual Studio and find when and where your problem occurs while running your code.

**Reply by zhangxiaopengY** [reply]
Yeah，I have solved it, thanks a lot.

**Reply by zhangxiaopengY** [reply]
Hi, @murphy-john , can we apply the prescribed velocity with file on version 2022? I find that there is a subroutine (ReadPrescribedAccelerationOrVelocityFromFile) in the source code, but when I used a PVF file to prescribe the velocity of a rigid entity, it didn't move with the time, however, the Moving Mesh can be observed with the prescribed velocities.

**Reply by Xie1185** [reply]
Hi, can you leave a contact information ，I am also a lover of Anura.

**Reply by 6cs4u** [reply]
I also encountered the problem of only grid movement. How did you solve it

**Reply by sunny12138-9527** [reply]
> Yeah，I have solved it, thanks a lot.是的，我已经解决了，非常感谢。

I am a master's student working on landslide simulations using Anura3D. My model consists of a bedrock layer and a landslide mass. However, I encounter an issue where particles at the toe of the slope fly out during the simulation. I suspect it might be related to an incorrect definition of the contact normal vector on the slope surface. The problem is, I’m not sure how to accurately locate the node positions in Anura3D to check and correct the normal vector definition.

This issue is critical to my research, and I would really appreciate any advice from those who have experience with Anura3D. If you have encountered similar particle flying issues or know how to resolve this, please share your insights.

Thank you very much for your help!

**Comment by ZeidBit** [comment]
Hey everyone, 

Glad to see more people are interested in slope run out simulations using Anura3D! I hope we can all collaborate to better understand how this can be achieved with MPM methods. I have a question regarding my landslide model: 

As part of my slope run-out analysis I am trying to simulate a landslide. After the initialization of the stresses phase under quasi-static convergence in the CPS 002 file I changed the time steps, lowered the dampening and set the contact formulation to slide along the linear elastic base. However, when running the calculation it stops at reading contact data from GOM file where it has a >PAUSE message displayed after the filename. Could I be describing the contact wrongly? If there is no error, why is it "pausing" with no solution? I attached a word file with screenshots of my model and set conditions. Any help would be much appreciated! 

[GitHub_Question.docx](https://github.com/Anura3D/Anura3D_OpenSource/files/8532534/GitHub_Question.docx)

**Reply by ayerrocolom** [reply]
I would highly recommend you to debug your problem to find at which line when reading the GOM file the code breaks. 
What kind of simulation are you doing? fully coupled? undrained? I am asking this because the names of the materials are "dry.., saturated..., unsaturated..." the combination of different types of formulations might create some issues.

**Comment by zhangxiaopengY** [comment]
Hey, everyone, can you give me some advice on how to set the absorbing boundary conditions in Anura 3D?   A reference material is required when set the absorbing boundary, how to set it, a new material need to be set here??? If need, how to set the geometry, material parameter, fixities...Thanks a lot for your suggestions.

**Reply by ayerrocolom** [reply]
The reference material is usually the material next to the boundary.

**Comment by Sukrity95** [comment]
I am using GID v15. And for Anura 3D analysis, Material tab is not showing up for defining material property, Even pressing Ctrl+A does not help to pop up the material tab. I have tried by reinstalling entire package.
This problem started when I drag down the opened pop up of material and it all of sudden hided out in task bar.
![screenshot](https://user-images.githubusercontent.com/104423208/165261289-9bc8ba0f-14ff-408c-8b3b-9e65670e4686.jpg)
.

**Reply by ayerrocolom** [reply]
If you click on the Anura3D menu, the Material tap should appear. Can you check and let us know? Thanks!

**Reply by Sukrity95** [reply]
Yes. It did appear but while clicking the material menu did not pop up in the window. Then I installed GiD v14 and I am using it. But the problem of GiD v15 is still not solved. I am waiting for GiD v16 to upgrade directly :) :)

**Comment by ghost** [comment]
I 'm using Anura3D to simulate the construction of tunnel which involved the installation of segment. During the calculation, i guess the soil material can't be changed to concrete. In Anura3D, can i change the material or relocate the material point during the calculation?

**Reply by FrancescaCeccato** [reply]
In Anura, you cannot change the material, but if the material has the same consitutive model you can change the properties. So one possibility is to create a new material, you can call it soil-concrete, start with soil properties and then switch to concrete properties later on in the simulation by using the CPS flag $APPLI_MATERIAL_UPDATE. If this flag is set to 1, the code will read again the material properties (gamma, young modulus, friction angle...) and reassign them to the materials.

**Reply by ghost** [reply]
Yeah, i got it. Thanks for your help!

**Comment by ghost** [comment]
I made a model to simulate a deep excavation. In this model, I simulate retaining structures and contact between soil and wall. But when the soil was excavated, the soil behind seemed penetrate the wall. Whether the parameters of contact was inappropriate accounting for this problem or other error?
![1653147960(1)](https://user-images.githubusercontent.com/105859708/169659173-81409a6b-b15d-4869-bf07-0acf30805c45.png)
![image](https://user-images.githubusercontent.com/105859708/169659202-e967d880-a60c-49c6-ba22-e7292312b90f.png)

**Reply by ayerrocolom** [reply]
The contact algorithm should be applied between two materials, one of them should not move or deform with respect to the mesh. In your case, both materials move, and the contact is not well defined. If you want to simulate this problem, you cannot use the contact algorithm.

**Reply by yfun0815** [reply]
Can i simulate by switching on contact and swithing off wall movement for several steps and then switching off contact and switching on wall movement for several steps? Thereby simulating the problem approximately.

**Reply by ghost** [reply]
In the simulation, the particle abruptly had a large velocity and stress, then the wall collapse. The wall was simulated by elastic material, it seems don't have failure criteria. What error arouse this problem?
![image](https://user-images.githubusercontent.com/105859708/169678140-7b7b174c-1678-4c8f-998f-e1ee9898a78e.png)
![image](https://user-images.githubusercontent.com/105859708/169678204-7862c77a-71f7-46cb-ad56-14cf9f6c316a.png)

**Reply by ayerrocolom** [reply]
In this problem, you cannot use the contact algorithm because both the wall and the soil deform. Note that the contact nodes are initialized based on the initial geometry and do not update during the simulation.

**Reply by ghost** [reply]
Yeah, i have abopted your advice and deleted the contact, but the above problem emerged. The acclerated praticle was like a bullet and destroyed the model.

**Reply by ghost** [reply]
This problem was solved by changing to use structured mesh. Maybe a pointed mesh will result in this problem. 
Thanks for your advice!

**Reply by Sukrity95** [reply]
@yfun0815 I also face the same problem. Can you please explain in details how to use structured mesh ? Thanks a lot

**Reply by ghost** [reply]
> @yfun0815 I also face the same problem. Can you please explain in details how to use structured mesh ? Thanks a lot

When i reduce the timestep, i found when the praticles gathered at a mesh point, they would explode. I guess the small angle of mesh at this point resulted in this problem. Therefore, I just made structured mesh and created standard 45 degree triangle shape of mesh. When you created mesh, you can choose the structured mesh in GID.

**Reply by Sukrity95** [reply]
Thank you very much @yfun0815. It really worked.

**Comment by Sukrity95** [comment]
![Error](https://user-images.githubusercontent.com/104423208/170798333-e5c4e1bc-eb59-47ad-b75e-286ae13ae54a.jpg)
Hi all, My simulation stops with the message "Maximum number of recursions reached". Please guide me to avoid this error.
Thanks a lot

**Reply by Xie1185** [reply]
I also encountered this problem. Have you solved it？

**Reply by murphy-john** [reply]
This error occurs when the code cannot find a new element after the position of a Material Point is updated. I don't have much experience with this issue but you could use the debug feature in Visual Studio to locate which Material Point is causing the error and find out why it happened. 

I recall having this issue in some of my models. I _think_ it was caused by a boundary condition not applied correctly, so it may be worth reviewing your boundary conditions.

**Comment by FrancescaCeccato** [comment]
This error mens that the algorithm could not find an element for a moving material point, maybe it is moving too fast (instability issue) or boundary conditions are not correctly applied or there is an issue with the geometry. What is your last output?

**Reply by Sukrity95** [reply]
I am trying to simulate the slope instability analysis with the prescribed velocity. And it was seen that particles start flying after step-10 and process stops ultimately at step-13 with error message.
Referring to below image, I use the parameters as:
Prescribed velocity - 0.05m/sec
Time step - 0.5 sec
No. of step - 30
Please suggest reduction in prescribed velocity may solve the problem or is there any other resorts.
![image](https://user-images.githubusercontent.com/104423208/170948440-7e0a5698-efbc-47d6-ad80-76147311f60f.png)

**Reply by FrancescaCeccato** [reply]
How did you apply the prescribed velocity? at MP or at the nodes? Are you using contact?
Note that if prescribed velocity is applied at the nodes or contact algorithm is used, then you must also use the moving mesh in order to keep the nodes always alinged with the material. Try different options to see what works better for this specific case. Probably prescribed velocity at MP and no contact is better.

**Reply by Sukrity95** [reply]
This is the update to my previous problem which i tried in last couple of weeks. I resolved the problem of particle fly problem, but I could not solve the "Max number of recursion reached".
I had tried prescribed velocity with MP and with nodes. Additionally, reduced the velocity and courants number but could not get the further stain in pile stabilized slope case.
Now I am trying to achieve the objective with application of seepage or pore water pressure.
Thanks

**Comment by Xie1185** [comment]
Hi everyone,

My name is Xie,I remember that Anura3D 2019.1 does not allow liquid and solid material simulated together, so I want to if Anura3D 2022 can solve this problem.

**Reply by FrancescaCeccato** [reply]
Hi! This new version do have the dounble point formulation available, but it is still under development. We'll be very happy to get your feedback on it and eventual improvements you'll implement!

**Comment by yuguzhangwu** [comment]
Hi everyone!
I wanted to simulate particle vibrations, similar to an earthquake. But I only know the constraints of fixed displacement. How can I apply a variable displacement condition? For example, a sinusoidal one.
Thanks a lot!

**Reply by Xie1185** [reply]
Can you leave a contact information ? We can communicate.

**Reply by yuguzhangwu** [reply]
OK. How about my email? 932606673@qq.com

**Reply by murphy-john** [reply]
Hello, applying variable boundary conditions is an active area of development by our community. We don't have this option available in the release. I will mention your question at our next developers meeting to see if there is a better answer.

**Reply by yuguzhangwu** [reply]
OK. Thanks for your reply!

**Comment by yuguzhangwu** [comment]
Hello everyone!
Could you please tell me how to apply a variable velocity(depending with time, such as a sinusoidal one) to the nodes in Anura3D?

**Reply by FrancescaCeccato** [reply]
You can apply a prescribed velocity to nodes or material points and then modify the subrotuine "UpdateMultipliersForTimeDependency" according to your special need. The Variable "CalParams%...%VelocitySolid..." is the one used for prescribed velocity.

**Reply by yuguzhangwu** [reply]
Thanks a lot! And could please tell me where "UpdateMultipliersForTimeDependency" is?

**Reply by FrancescaCeccato** [reply]
It is in MPMDynamicExplict.for, but I suggest you to use the function "find" of visualStudio (Crtl+F) which is very handy for finding subroutines in the code.

**Reply by yuguzhangwu** [reply]
Thank you very much! I have found it with your help. So should I compile the code again?

**Reply by FrancescaCeccato** [reply]
yes, when you cange something you have to compile again and use the new executable for the simulations.

**Reply by yuguzhangwu** [reply]
OK! Let me have a try. Many thanks!

**Reply by yuguzhangwu** [reply]
Hello. I'm sorry I was still puzzled because of my poor coding ability. How could this subroutine work? Should I need to it in somewhere?

**Reply by yuguzhangwu** [reply]
Hello. I'm sorry I was still puzzled because of my poor coding ability. How could this subroutine work? Should I need to call it in somewhere?

**Reply by FrancescaCeccato** [reply]
This subroutine updates the multipliers for the current time step and it is called at the beginning of the time step. If you want to change the multiplier you have to modify this subroutine. Maybe you can add a few lines of code in the subroutine to adjust the multiplier simulating a sinusoidal load. Alternatively you can add a reading from file of this multiplier. This depends on your objective...

**Reply by yuguzhangwu** [reply]
Thank you so much! Sorry for disturbing you again. I have one more question: So am I only capable to modify a multiplier or I could also determine a function of time and velocity based on the prescribed velocity? If I could, which variable represent the prescribed velocity  input in GiD?

**Reply by sunbenbo** [reply]
Dear Francesca Ceccato

I have read your reply about the seismic load to all in GITHUB, but there's just mentioned "UpdateMultipliersForTimeDepencency ()".How exactly does the program read the file, directly read the file TXT or? Can you share a case?

**Reply by FrancescaCeccato** [reply]
I'm not sure I understood correctly your question, but you should check the subroutine ReadPrescribedAccelerationOrVelocityFromFile().
The file is a txtx with 2 colums: time & velocity.

**Reply by sunbenbo** [reply]
If I have a txt with two columns of data, namely time and velocity/acceleration, as follows：
5
0.01 0.1
0.02 0.2
0.03 0.5
0.04 0.7
0.05 0.8
How should I read this txt below, and at the same time how to apply the Prescribed velocity you emphasized?

**Reply by AlsardiVirginiaTech** [reply]
Hello. You can directly read your text file without specifying "5" lines. You file needs to have the extension '.PVF'.  Also, the multiplier needs to be specified from 'file'. You also need to input the name of that file under $$PRESCRIBED_VELOCITY flag. Please check out the code below $$PRESCRIBED_VELOCITY so that you can see how file multipliers are specified. The prescribed velocity node IDs are specified as usual in the GOM file. Let me know if you have more questions.

**Reply by AlsardiVirginiaTech** [reply]
Depending on your application, I recommend that you use the moving mesh when you are applying prescribed velocity from file to ensure that the material points remain within the computational mesh.

**Comment by Xie1185** [comment]
I use dounble point formulation,but&nbsp;&nbsp;it show a error as this:


what it's meaning,what I can do?


安徽&nbsp;解先争
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Anura3D/Anura3D_OpenSource"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月13日(星期一) 下午3:48
***@***.***&gt;;
抄送:&nbsp;"安徽 ***@***.******@***.***&gt;;
主题:&nbsp;Re: [Anura3D/Anura3D_OpenSource] Welcome to Anura3D_OpenSource Discussions! (Discussion #1)





 
Hi! This new version do have the dounble point formulation available, but it is still under development. We'll be very happy to get your feedback on it and eventual improvements you'll implement!

**Reply by murphy-john** [reply]
Hello, it appears you images or text did not load properly. Will you reload the images or explain the error with text?

**Comment by yuguzhangwu** [comment]
Hello everyone!
I have a question about contact. Can I use two types of materials both of which could move(free) in contact problems?

**Reply by murphy-john** [reply]
[Please see this comment.](https://github.com/Anura3D/Anura3D_OpenSource/discussions/1#discussioncomment-2797109)

**Reply by yuguzhangwu** [reply]
OK. Thank you!

**Comment by yuguzhangwu** [comment]
Hello everyone！
I try to apply a contact formulation in column column collapse problem just like shown in tutorial manual. And the following picture is my settings. But the result was like a no-friction one(all granular went out). How could I apply contact correctly?
<img width="739" alt="微信图片_20220620151956" src="https://user-images.githubusercontent.com/106877284/174547165-50ab2b63-faa7-4b81-a4ef-31d28744b1ec.png">

**Reply by FrancescaCeccato** [reply]
Did you shitch the contact ON in the CPS file? $APPLY_CONTACT 1?

**Reply by yuguzhangwu** [reply]
Yes, I did.

**Reply by FrancescaCeccato** [reply]
This is strange, did you try to debug your problem a check if the variables are all read correctly?

**Reply by yuguzhangwu** [reply]
Thanks, I have found the reason that I defined the column material fist. But the result was still strange that the base was just like a rigid body,  when the column material got to the base, it will "jump up" just like the following picture. And the calculation will break off. 
![1657624909871](https://user-images.githubusercontent.com/106877284/178478605-b0d53c10-5fc9-4338-b3d0-6a8e8a571d92.png)

**Reply by yuguzhangwu** [reply]
![1657624909871](https://user-images.githubusercontent.com/106877284/178479425-caf75ec7-ce2a-43c7-a9ac-0c66026d0d25.jpg)

**Reply by Shachen75** [reply]
Hi, did you solve this particle fly problem? It is supposed to define the base material in the last step and column material is defined first. Can you please double check?

**Reply by sunny12138-9527** [reply]
> Hi, did you solve this particle fly problem? It is supposed to define the base material in the last step and column material is defined first. Can you please double check?您好，您解决粒子飞行问题了吗？它应该在最后一步定义基础材质，而柱材质是先定义的。您能再确认一下吗？

I am a master's student working on landslide simulations using Anura3D. My model consists of a bedrock layer and a landslide mass. However, I encounter an issue where particles at the toe of the slope fly out during the simulation. I suspect it might be related to an incorrect definition of the contact normal vector on the slope surface. The problem is, I’m not sure how to accurately locate the node positions in Anura3D to check and correct the normal vector definition.

This issue is critical to my research, and I would really appreciate any advice from those who have experience with Anura3D. If you have encountered similar particle flying issues or know how to resolve this, please share your insights.

Thank you very much for your help!

**Comment by zhangxiaopengY** [comment]
Hi, everyone, the unit of the value stored in the STRESS array is Kpa, is that right?

**Reply by FrancescaCeccato** [reply]
Yes

**Comment by Shachen75** [comment]
Hi, how to simulate a seismic loading in Anura3D? Can we apply a series of prescribed  velocities at the model base?  Thanks.

**Reply by FrancescaCeccato** [reply]
Yes, this is possible. You can modify the code to read the multiplier of prescribed velocity from a file. Check the subroutine "UpdateMultipliersForTimeDepencency()"

**Reply by Shachen75** [reply]
Thanks!

**Reply by CPLA81192** [reply]
Hi, I have a Prescribed velocity file in the project folder, but the program shows "Prescribed velocity file does not exist!", Why does this error occur?
![1701936270439](https://github.com/Anura3D/Anura3D_OpenSource/assets/97551917/afa80739-4f7a-4320-a70d-7f851925834b)
![1701936286763](https://github.com/Anura3D/Anura3D_OpenSource/assets/97551917/051147b9-26e5-43c2-8186-d2b480ab0cd1)

**Reply by FrancescaCeccato** [reply]
I suggest to debug and check where the code stop what is the name of the file it is looking for, there might be a small mistake there.

**Reply by AlsardiVirginiaTech** [reply]
Hello all, make sure you have the file name typed in the CPS file under the $$PRESCRIBED_VELOCITY flag. I am certain that is your problem as it is showing in your screen shot *empty*.PVF. The code is not recognizing this empty file name. Best wishes.

**Reply by CPLA81192** [reply]
Thank you! the file name does not meet the requirements, it has been solved

**Reply by CPLA81192** [reply]
Hi! I want to know How to set PrescribedAcceleration from file?

**Reply by kuailebozai** [reply]
@CPLA81192 Hello, may I ask how this ". PVF" file was read? I am facing the same problem as you, what should I do? Thank you very much

**Comment by sunxy379** [comment]
Hello, everyone!

I would like to modify the properties of materials with the change of space, like the VUSDFLD subroutine in the ABAQUS. Can you give me some info about this feature or which subroutine I should check?

Best regards,
Sunrise

**Comment by FrancescaCeccato** [comment]
With the current version this is not very eary, it depends how you want to change properties. Material properties are stored in the vector MatParams(materialID) at the level of MaterialID. State variables are stored at the material point level in the vector ESMstatevArray(MaterialPointID,:). This means that changing spatially the state variables is relatively easy, you can modify the subroutine ReadMaterialParameters(), in contrast changing material parameters is complicated because you need a material for each set of material parameters.
The solution is to store material parameters at the material point level, but this reduces the computational efficiency of the code.

**Reply by sunxy379** [reply]
Thank you very much for taking the time to respond to my issue, I'll have a try.

**Comment by CPLA81192** [comment]
Hi! I want to know how to apply PrescribedAcceleration from file?

**Reply by AlsardiVirginiaTech** [reply]
Hello. In MPM, strains are calculated from velocity fields. Applying an only-acceleration prescription will not enforce the boundary condition. Due to this, we do not currently have a prescribed acceleration. I encourage you to baseline correct your ground motion in acceleration, integrate to velocity and prescribe it. I have a branch where this happens within the code (i.e., you input your ground motion in acceleration, and we do the integration internally in Fortran).  This will be released in the next sprint. You can reply here if you have more questions.

**Reply by AlsardiVirginiaTech** [reply]
Instead of 'linear'/'step' multipliers, input 'file' and include the file name at the end of the line.

**Reply by CPLA81192** [reply]
Thank you for your reply. I will try it at the specified prescribed velocity

**Reply by kuailebozai** [reply]
@CPLA81192 Hello, may I ask if there has been any progress in using loading speed to simulate seismic loads? I would also like to analyze this issue. Thank you very much.

**Comment by weilinlee0308** [comment]
<img width="1008" alt="image" src="https://github.com/user-attachments/assets/5a0d89a5-29e0-418f-a1fd-8c4f1142729d">
Here is the code for the DSigWP in the condition of unsaturated soil
<img width="605" alt="image" src="https://github.com/user-attachments/assets/1471d895-71c6-4585-abc3-0c1604818a5b">
Here is the equation for the DSigWP in the condition of unsaturated soil. Accordingly, should we change " + WaterAdvectiveFlux" to " - WaterAdvectiveFlux"? It seems that "(Sr*WaterDensity)*DEpsVol + WaterAdvectiveFlux" does not match the formula. Am I correct?

**Reply by FrancescaCeccato** [reply]
Note that the equation in the code is the opposite of 3.37 because the minus sign is in lambda and the water advective flux is calcualted using v_L-v_s (opposite of eq.3.37).

**Reply by weilinlee0308** [reply]
Thanks!!!
